org.metaborg.spoofax.core/src/main/java/org/metaborg/spoofax/core/dynamicclassloading/BuilderInput.java

```diff
@@ -0,0 +1,88 @@
+package org.metaborg.spoofax.core.dynamicclassloading;
+
+import javax.annotation.Nullable;
+
+import org.apache.commons.vfs2.FileObject;
+import org.metaborg.util.resource.ResourceUtils;
+import org.spoofax.interpreter.terms.IStrategoTerm;
+import org.spoofax.terms.StrategoString;
+import org.spoofax.terms.StrategoTuple;
+import org.spoofax.terms.TermFactory;
+
+public class BuilderInput extends StrategoTuple implements IBuilderInput<IStrategoTerm, IStrategoTerm> {
+    private final IStrategoTerm selection;
+    private final IStrategoTerm position;
+    private final IStrategoTerm ast;
+    private final @Nullable FileObject resource;
+    private final @Nullable FileObject location;
+
+    @SuppressWarnings("deprecation")
```

Comment:
What was deprecated that you needed to suppress the warning?

Reply:
`TermFactory.EMPTY_LIST`. Completely unclear why it is deprecated, and was never removed.

Reply:
That would change the API a little because you'd need to pass a factory. I'm not too worried about mutable attachments on the list of annotations of the builder tuple. But I can change it for the sake for consistent code style if you like.